### PR TITLE
Add test of common file

### PR DESCRIPTION
**Pull request by Mickael-Martin**
To add a check for the present of the source of _common.sh file. Check if duplicate, absent or not correct (execution context).
Ref https://github.com/YunoHost/package_linter/issues/50

### CODE REVIEW

File: package_linter.py [hunk 1]
```diff
@@ -437,6 +437,26 @@ def check_deprecated_practices(script):
     if script["name"] == "install" and "ynh_print_info" not in script["shlex"] and "ynh_script_progression" not in script["shlex"]:
         print_warning("Please add a few messages for the user, to explain what is going on (in friendly, not-too-technical terms) during the installation. You can use 'ynh_print_info' or 'ynh_script_progression' for this.")
 
+def check_source_common(script):
+    #test for standard execution
+    if script["name"] in ["install","upgrade","remove"]:
+        count_common=len(re.findall("^source _common.sh",script["raw"],flags=re.MULTILINE))
```
Comment (kay0u) [inline package_linter.py]: `source ./_common.sh` is valid too in this case

Comment (Mickael-Martin) [reply package_linter.py]: If you want, there are many syntax for calling the right file. I add you suggestion.

Comment (Mickael-Martin) [reply package_linter.py]: done

File: package_linter.py [hunk 2]
```diff
@@ -437,6 +437,26 @@ def check_deprecated_practices(script):
     if script["name"] == "install" and "ynh_print_info" not in script["shlex"] and "ynh_script_progression" not in script["shlex"]:
         print_warning("Please add a few messages for the user, to explain what is going on (in friendly, not-too-technical terms) during the installation. You can use 'ynh_print_info' or 'ynh_script_progression' for this.")
 
+def check_source_common(script):
+    #test for standard execution
+    if script["name"] in ["install","upgrade","remove"]:
+        count_common=len(re.findall("^source _common.sh|^source ./_common.sh",script["raw"],flags=re.MULTILINE))
+        if count_common == 0:
+            print_warning("Calling _common.sh seams not present in this script, please add \"source _common.sh\"")
```
Comment (alexAubin) [inline package_linter.py]: Uh but is it expected / wanted to have a _common.sh for all apps, even if you don't need such a file ?

Comment (maniackcrudelis) [reply package_linter.py]: No, it's not.

Comment (alexAubin) [reply package_linter.py]: So I fixed / simplify the check ...

**Comment by alexAubin** [comment]
Planning to merge in a few days ... any feedback welcome

**Comment by alexAubin** [comment]
Yolomergin' (still feel free to complain if you see an issue with this PR ;^p)

**Comment by maniackcrudelis** [comment]
To print a warning about that, why not. But do not pop an error for that test.
If a _common.sh isn't sourced correctly, bash will failed, which is enough to raise an error.

But to use `source ./_common.sh` isn't always an error.
In [Discourse](https://ci-apps.yunohost.org/ci/job/2402), there's no error about it, https://github.com/YunoHost-Apps/discourse_ynh/blob/master/scripts/backup#L8-L13

**Comment by alexAubin** [comment]
@maniackcrudelis : Should be fixed by https://github.com/YunoHost/package_linter/commit/9181e08814182c6db686296de48524a24831bec6

**Pull request merged**